File: files/2025-11-09/CSEC 09112025.R
library(tidyverse)
library(readxl)

path = "2025-11-09/Challenge 74.xlsx"
input = read_excel(path, range = "B3:F8", col_names = FALSE)
test  = read_excel(path, range = "H3:K15",  col_names = FALSE)

result = input %>%
  mutate(aspect = ifelse(str_detect(...1, "(?i)item", negate = TRUE), ...1, NA_character_)) %>%
  fill(aspect) %>%
  filter(...1 != aspect) %>%
  pivot_longer(cols = where(is.numeric), names_to = "item", values_to = "value") %>%
  select(...1, ...2, ...3 = aspect, ...4 = value) %>%
  mutate(...4 = as.character(...4)) %>%
  add_row(...1 = NA_character_,
          ...2 = NA_character_,
          ...3 = NA_character_,
          ...4 = "Values",
          .before = 1)

#provided answer is not correct.

File: files/2025-11-09/CSEC 09112025.py
import pandas as pd
import numpy as np

path = "2025-11-09/Challenge 74.xlsx"
input = pd.read_excel(path, usecols="B:F", skiprows=2, header=None, nrows=6)
test = pd.read_excel(path, usecols="H:K", skiprows=2, header=None, nrows=13)

result = (
    input
    .assign(
        aspect=lambda df: np.where(
            ~df.iloc[:, 0].str.contains("(?i)item", regex=True),
            df.iloc[:, 0],
            np.nan
        )
    )
    .ffill()
    .loc[lambda df: df.iloc[:, 0] != df["aspect"]]
    .melt(
        id_vars=[input.columns[0], input.columns[1], "aspect"],
        var_name="item",
        value_name="value"
    )
    .loc[:, [input.columns[0], input.columns[1], "aspect", "value"]]
    .assign(value=lambda df: df["value"].astype(str))
)
header = pd.DataFrame({
    result.columns[0]: [np.nan],
    result.columns[1]: [np.nan],
    result.columns[2]: [np.nan],
    result.columns[3]: ["Values"]
})
result = pd.concat([header, result], ignore_index=True)

# provided answer is not correct